src/planner/binder/expression/bind_operator_expression.cpp

```diff
@@ -121,18 +121,10 @@ BindResult ExpressionBinder::BindExpression(OperatorExpression &op, idx_t depth)
 			function_name = "json_extract";
 			// Make sure we only extract array elements, not fields, by adding the $[] syntax
 			auto &i_exp = BoundExpression::GetExpression(*op.children[1]);
-			if (i_exp->GetExpressionClass() == ExpressionClass::BOUND_CONSTANT &&
-			    !i_exp->Cast<BoundConstantExpression>().value.IsNull()) {
+			if (i_exp->GetExpressionClass() == ExpressionClass::BOUND_CONSTANT) {
 				auto &const_exp = i_exp->Cast<BoundConstantExpression>();
-				if (const_exp.value.TryCastAs(context, LogicalType::UINTEGER)) {
-					// Array extraction: if the cast fails it's definitely out-of-bounds for a JSON array
-					auto index = UIntegerValue::Get(const_exp.value);
-					index -= index > 0; // Subtract 1 for SQL 1-based indexing (except when accessing from back)
```

Comment:
We should keep the new bind code, just remove this line. Then we can still support the `['field']` extraction syntax